app/services/xcode_manager_service.rb

```diff
@@ -67,7 +67,9 @@ def xcode_queue
     #
     # the resulting list is sorted with the most recent release as first element
     def available_xcode_versions
-      return installer.seedlist.reverse
+      apple_id_credentials_block do
+        return installer.seedlist.reverse
```

Comment:
I would like to discourage the use of `return` inside blocks. They can lead to surprising behavior. Correct me if I am wrong, but this will not reach the `ENV.delete` methods in the `ensure` part of the block.

Reply:
Ohhh right, nice catch!!